examples/torch/dqn_atari.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python3
+"""An example to train a task with DQN algorithm.
+
+Here it creates a gym environment CartPole, and trains a DQN with 50k steps.
+"""
+import click
+import gym
+import numpy as np
+import psutil
+import torch
+
+from garage import wrap_experiment
+from garage.envs import GymEnv
+from garage.envs.wrappers.clip_reward import ClipReward
+from garage.envs.wrappers.episodic_life import EpisodicLife
+from garage.envs.wrappers.fire_reset import FireReset
+from garage.envs.wrappers.grayscale import Grayscale
+from garage.envs.wrappers.max_and_skip import MaxAndSkip
+from garage.envs.wrappers.noop import Noop
+from garage.envs.wrappers.resize import Resize
+from garage.envs.wrappers.stack_frames import StackFrames
+from garage.experiment.deterministic import set_seed
+from garage.np.exploration_policies import EpsilonGreedyPolicy
+from garage.replay_buffer import PathBuffer
+from garage.sampler import FragmentWorker, LocalSampler
+from garage.torch import set_gpu_mode
+from garage.torch.algos import DQN
+from garage.torch.policies import DiscreteQFArgmaxPolicy
+from garage.torch.q_functions import DiscreteCNNQFunction
+from garage.trainer import Trainer
+
+hyperparams = dict(n_epochs=500,
+                   steps_per_epoch=20,
+                   sampler_batch_size=500,
+                   lr=1e-4,
+                   discount=0.99,
+                   min_buffer_size=int(1e4),
+                   n_train_steps=125,
+                   target_update_freq=2,
+                   buffer_batch_size=32,
+                   max_epsilon=1.0,
+                   min_epsilon=0.01,
+                   decay_ratio=0.1,
+                   buffer_size=int(1e4),
+                   hidden_sizes=(512, ),
+                   hidden_channels=(32, 64, 64),
+                   kernel_sizes=(8, 4, 3),
+                   strides=(4, 2, 1),
+                   clip_gradient=10)
+
+
+@click.command()
+@click.argument('env', type=str)
+@click.option('--seed', default=24)
+@click.option('--n', type=int, default=psutil.cpu_count(logical=False))
+@click.option('--buffer_size', type=int, default=None)
+@click.option('--max_episode_length', type=int, default=None)
+def main(env=None,
```

Comment:
this doesn't seem like canonical usage of wrap_experiment -- there a reason you set it up like this?

@krzentner PTAL

Reply:
This is kind of a weird setup, especially with the `hyperparams` dictionary, but it's something I could reasonably see myself writing. I also think having a general dqn atari example like this is valuable (instead of an example for a single atari environment, or many atari environment examples). That kinda necessitates some of this complexity, so it's fine.

Reply:
It's just easier to run various atari experiments using one launcher. The extra click arguments are just so that tests can set a smaller buffer size and max episode length.